File: mobile-client/src/pages/route-view/route.ts
import { Point } from './point';
export class Route{

    constructor(public name: string, public points: Point[] ) {  
    }

    getName() {
        return this.name;
    }

    setName(name) {
        this.name = name;
    }

    addPoint(point: Point) {
        this.points.push(point)   
    }

    removePoint(point) {
        for (let i = 0; i < this.points.length; i++){
            if (this.points[i] == point) {
                this.points.splice(i, 1);
            }
        }
    }    
}

File: profile-service/src/main/java/org/vvasiloud/service/profile/domain/Route.java
package org.vvasiloud.service.profile.domain;

import org.springframework.data.annotation.Id;
import org.springframework.data.geo.Point;

import javax.validation.constraints.NotNull;
import java.util.List;

public class Route {

    @Id
    private String name;

    @NotNull
    private List<Point> points;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Point> getPoints() {
        return points;
    }

    public void setPoints(List<Point> points) {
        this.points = points;
    }
}
